File: modules/facade/src/test/java/com/teamwizardry/librarianlib/facade/example/containers/DirtSetterItem.java
package com.teamwizardry.librarianlib.facade.example.containers;

import com.teamwizardry.librarianlib.facade.example.ExampleModContainers;
import net.minecraft.entity.player.ServerPlayerEntity;
import net.minecraft.item.Item;
import net.minecraft.item.ItemUseContext;
import net.minecraft.util.ActionResultType;
import net.minecraft.util.text.TranslationTextComponent;
import org.jetbrains.annotations.NotNull;

public class DirtSetterItem extends Item {
    public DirtSetterItem(@NotNull Item.Properties properties) {
        super(properties);
    }

    @Override
    public ActionResultType useOn(ItemUseContext context) {
        if (!context.getLevel().isClientSide) {
            ExampleModContainers.dirtSetterContainerType.open(
                    (ServerPlayerEntity) context.getPlayer(),
                    new TranslationTextComponent("modid.container.dirt_setter"),
                    // additional constructor arguments:
                    context.getClickedPos()
            );
        }
        return ActionResultType.SUCCESS;
    }
}

File: modules/foundation/src/test/kotlin/com/teamwizardry/librarianlib/foundation/testmod/customtypes/DirtSetterItem.kt
package com.teamwizardry.librarianlib.foundation.testmod.customtypes

import com.teamwizardry.librarianlib.foundation.item.BaseItem
import com.teamwizardry.librarianlib.foundation.testmod.ModContainers
import net.minecraft.item.ItemUseContext
import net.minecraft.util.ActionResultType
import net.minecraft.entity.player.ServerPlayerEntity
import net.minecraft.util.text.TranslationTextComponent

class DirtSetterItem(properties: Properties): BaseItem(properties) {
    override fun useOn(context: ItemUseContext): ActionResultType {
        if (!context.level.isClientSide) {
            ModContainers.dirtSetter.open(
                (context.player as ServerPlayerEntity?)!!,
                TranslationTextComponent("librarianlib-foundation.test_container.title"),
                context.clickedPos
            )
        }
        return ActionResultType.SUCCESS
    }
}